src/netcam_rtsp.c

```diff
@@ -305,13 +305,68 @@ static int netcam_decode_vaapi(struct rtsp_context *rtsp_data)
             return retcd;
         }
         rtsp_data->frame->format=AV_PIX_FMT_YUV420P;
+
+        retcd = av_hwframe_transfer_data(rtsp_data->frame, hw_frame, 0);
+        if (retcd < 0) {
+            MOTION_LOG(INF, TYPE_NETCAM, NO_ERRNO
+                ,_("%s: Error transferring HW decoded to system memory")
+                ,rtsp_data->cameratype);
+            my_frame_free(hw_frame);
+            return -1;
+        }
+
+        my_frame_free(hw_frame);
+
+        return 1;
+    #else
+        (void)rtsp_data;
+        return 1;
+    #endif
+}
+
+static int netcam_decode_cuda(struct rtsp_context *rtsp_data)
+{
+
+    #if ( MYFFVER >= 57083)
+
+        int retcd;
+        char errstr[128];
+        AVFrame *hw_frame = NULL;
+
+        hw_frame = my_frame_alloc();
+
+        retcd = avcodec_receive_frame(rtsp_data->codec_context, hw_frame);
+        if ((rtsp_data->interrupted) || (rtsp_data->finish) || (retcd < 0) ){
+            if (retcd == AVERROR(EAGAIN)){
+                retcd = 0;
+            } else if (retcd == AVERROR_INVALIDDATA) {
+                MOTION_LOG(INF, TYPE_NETCAM, NO_ERRNO
+                    ,_("%s: Ignoring packet with invalid data")
+                    ,rtsp_data->cameratype);
+                retcd = 0;
+            } else if (retcd < 0) {
+                av_strerror(retcd, errstr, sizeof(errstr));
+                    MOTION_LOG(INF, TYPE_NETCAM, NO_ERRNO
+                        ,_("%s: Rec frame error: %s")
+                        ,rtsp_data->cameratype, errstr);
+                retcd = -1;
+            } else {
+                retcd = -1;
+            }
+            my_frame_free(hw_frame);
+            return retcd;
+        }
+        rtsp_data->frame->format=AV_PIX_FMT_NV12;
```

Comment:
Have you tried this with `rtsp_data->frame->format=AV_PIX_FMT_YUV420P;` instead?  

For VAAPI, when the format is specified here to allow for the HW decoder to directly transfer the image to the Motion internal pixel format.  

If the image comes back from HW in NV12 format, then Motion will have to do the conversion to YUV420P in software.

Reply:
Yea I tried with `AV_PIX_FMT_YUV420P` initially but the frame came back corrupted (green lines and overlap artifacts). I think cuda support `AV_PIX_FMT_YUV420P` format but maybe issue with my driver/setup

Reply:
`h264_cuvid` decoder doesn't support `AV_PIX_FMT_YUV420P` frame format, and only output `AV_PIX_FMT_NV12`

Reply:
```
Decoder h264_cuvid [Nvidia CUVID H264 decoder]:
    General capabilities: delay avoidprobe hardware 
    Threading capabilities: none
    Supported hardware devices: cuda 
    Supported pixel formats: cuda nv12 p010le p016le

Decoder hevc_cuvid [Nvidia CUVID HEVC decoder]:
    General capabilities: delay avoidprobe hardware 
    Threading capabilities: none
    Supported hardware devices: cuda 
    Supported pixel formats: cuda nv12 p010le p016le
```

Look like `AV_PIX_FMT_YUV420P` isn't support by cuda h264/h265 decoder :(